### ISSUE TITLE
Interrupt long songs

### ISSUE THREAD

**Issue by gammaw** [AzuraCast/AzuraCast#1051]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by mmmnmnm** [comment]
Hi, 

is it possible to define playlists so that a long song is interrupted by a new (higher weight) song? For example, if I have a default playlist playing between shows and I want that a specific show starts playing at a specific time, I don't want that some long song in the default playlist prevents the show from starting. Is it possible? 

Thanks, 

Peter

**Comment by SlvrEagle23** [comment]
@mmmnmnm At the moment, it's not possible to do this because of the way the AzuraCast AutoDJ works, where the next song is only requested at the end of the currently playing song.

The only way to allow this to happen would be to turn on "Manual AutoDJ" mode (which has Liquidsoap run its own AutoDJ settings internally) in the advanced settings, but even then it requires specific flags that are not applied to playlists by default.

In other words, it's possible but logistically it's difficult to integrate into the software.

**Comment by mmmnmnm** [comment]
@SlvrEagle23: thanks. We do need to fix it somehow. Where can I configure Liquidsoap in manual AutoDJ mode? How can I add those flags to playlists? It's different from advanced playlists right (https://github.com/AzuraCast/azuracast.com/blob/master/AdvancedPlaylists.md)?

**Comment by gammaw** [comment]
@SlvrEagle23 what exactly does above commit do (277c992053021d14f67064458af6cbbe7192aca6)? Would this commit make it possible to interrupt songs? If not, is it on your agenda to add this feature? We would love it! :)

**Comment by SlvrEagle23** [comment]
@gammaw Sorry, the commit message there appears to have been a typo referencing the wrong issue.

Currently, this issue isn't on my immediate agenda to resolve because of how Liquidsoap is configured under the hood to deliberately avoid interrupting songs in all cases except when a live DJ connects. I'm not confident that this could be overridden in a graceful way, or in a way that we could build into the software experience itself.

**Comment by gammaw** [comment]
@SlvrEagle23 I see. Is there a way to schedule that, starting at a specific time, every song in a playlist is played **exactly once**? The problem is that if a 59-minute playlist (for example) is scheduled between 6 and 7pm but one long track starts shortly before 7pm (played for the second time), then this track will run until it finishes.

**Comment by gammaw** [comment]
@SlvrEagle23 I guess I'll try to use custom Liquidsoap configurations. Can you tell me what the difference is between 1) using advanced playlists but disabling Manual AutoDJ Mode and 2) enabling Manual AutoDJ Mode? Can I use AzuraCast's music files and playlists (the ones uploaded and categorized on the AzuraCast web interface) with option 2)?
